### ISSUE TITLE
GPU octree error with big PointClouds. 

### ISSUE THREAD

**Issue by Goldesel23** [PointCloudLibrary/pcl#2008]
:warning: This is a issue tracker, please use our mailing list for questions: www.pcl-users.org. :warning: 

## Your Environment

* Operating System and version: Windows 10 Pro / Ubuntu 16.04
* Compiler:
* PCL Version: 1.8.1

## Expected Behavior

pcl::gpu::Octree should be able to build for different PointClouds. 

## Current Behavior

It works on some PointClouds but in other gives the following error: Error: unspecified launch failure       D:/Libraries/pcl-pcl-1.8.1/gpu/octree/src/cuda/octree_builder.cu:403

## Code to Reproduce

```cpp
pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZ>);
pcl::io::loadPCDFile("Original.pcd", *cloud);
pcl::gpu::Octree::PointCloud cloud_device;
cloud_device.upload(cloud->points);`
pcl::gpu::Octree octree_device;
octree_device.setCloud(cloud_device);
octree_device.build();
```

## Context

Althogh it fails with bigger pointClouds the test works with the provided tests of DataGenerator data;
data.data_size = 871000;

Has anyone faced this kind of problem?

**Comment by SergioRAgostinho** [comment]
Is the pointcloud you're using, uploaded somewhere so we can give it a try?

**Comment by Goldesel23** [comment]
[Original.txt](https://github.com/PointCloudLibrary/pcl/files/1397601/Original.txt)

Here is the pointCloud. I put it in a txt file because the pcd file upload did not work in the issue. Thanks for the help!

**Comment by SergioRAgostinho** [comment]
So I was not exactly able to reproduce your problem because I think it's something dependent on your hardware already. "Unspecified launch failure" happens normally when you try to launch kernel with incorrect grid/thread/warp sizes which your target hardware doesn't support. (Sorry for being imprecise with the lingo but I'm not experienced in CUDA)

I do get however a faulty memory access the moment I used 951 points of the cloud you provided, being caused by octree_builder.cu:246. Whatever the fix is for my problem, I'm almost certainly sure it won't solve yours.

On which hardware are you trying to run this? And what is your CUDA runtime version?

**Comment by Goldesel23** [comment]
Sorry for the late response, I have been using a GTXM960 with cuda 8.0. even with the Kernel with grid and block size of 1 it fails with the first 960 points (which does not happen in CPU version). I think the problem is in the algorithm. I spent some days on this, but was not able to figure out why it occurs. 

I think it might be due to the fact there are some outliers in those points and the algorithm gives an error in that case. Because if I run without those outliers, the algorithm runs smoothly. Sorry for not spotting exactly were the problem occured, but since it was for a work project, I had to move on with other tasks.
